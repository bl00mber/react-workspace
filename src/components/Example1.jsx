import PropTypes from 'prop-types';
import React from 'react';

const Example1 = ({
  counter,
  onIncrementCounter,
}) => (
  <div>
    Clicked&nbsp;
    {counter}
    &nbsp;time
    {counter !== 1 ? 's' : ''}
    <br />
    <button type="button" onClick={() => onIncrementCounter()}>
      Click me
    </button>
  </div>
);

Example1.propTypes = {
  counter: PropTypes.number.isRequired,
  onIncrementCounter: PropTypes.func.isRequired,
};

export default Example1;
