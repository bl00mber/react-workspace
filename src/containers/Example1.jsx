import { connect } from 'react-redux';

import Example1 from '@/components/Example1';
import { incrementCounter } from '@/store/actions';

const mapStateToProps = state => ({
  counter: state.counter,
});

const mapDispatchToProps = dispatch => ({
  onIncrementCounter: () => dispatch(incrementCounter()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Example1);
