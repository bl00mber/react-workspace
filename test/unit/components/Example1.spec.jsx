import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Example1 from '@/components/Example1';

describe('Example1', () => {
  test('should match enzyme shallow snapshot', () => {
    const props = {
      counter: 0,
      onIncrementCounter: () => { },
    };
    const wrapper = shallow(<Example1 {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should match enzyme shallow snapshot when counter is 1', () => {
    const props = {
      counter: 1,
      onIncrementCounter: () => { },
    };
    const wrapper = shallow(<Example1 {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should call onIncrementCounter clicking on button', () => {
    const props = {
      counter: 1,
      onIncrementCounter: jest.fn(),
    };
    const wrapper = shallow(<Example1 {...props} />);
    wrapper.find('button').at(0).simulate('click');
    expect(props.onIncrementCounter).toHaveBeenCalledWith();
  });
});
