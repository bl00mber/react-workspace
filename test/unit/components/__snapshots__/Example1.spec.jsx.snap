// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example1 should match enzyme shallow snapshot 1`] = `
<div>
  Clicked 
  0
   time
  s
  <br />
  <button
    onClick={[Function]}
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`Example1 should match enzyme shallow snapshot when counter is 1 1`] = `
<div>
  Clicked 
  1
   time
  <br />
  <button
    onClick={[Function]}
    type="button"
  >
    Click me
  </button>
</div>
`;
